### ISSUE TITLE
NavigationServer3D.BakeFromSourceGeometryData has a default third parameter but in C# the default parameter doesn't work.

### ISSUE THREAD

**Issue by viksl** [godotengine/godot#83349]
### Godot version

4.2 beta 1

### System information

Windows 11 - Vulkan - Nvidia RTX 4070 - intel i5 13600KF

### Issue description

The documentation says the `BakeFromSourceGeometryData` has a `Callable()` as a default third parameter, it works in gdscript (you can just use two paramaters) but in C# if you don't provide the third parameter (`Callable`) you will get an error when the code runs.

```
E 0:00:02:0226   Marshaling.cs:261 @ Godot.NativeInterop.godot_callable Godot.NativeInterop.Marshaling.ConvertCallableToNative(Godot.Callable&): System.NullReferenceException: Object reference not set to an instance of an object.
  <C# Error>     System.NullReferenceException
  <C# Source>    /root/godot/modules/mono/glue/GodotSharp/GodotSharp/Core/NativeInterop/Marshaling.cs:261 @ Godot.NativeInterop.godot_callable Godot.NativeInterop.Marshaling.ConvertCallableToNative(Godot.Callable&)
  <Stack Trace>  Marshaling.cs:261 @ Godot.NativeInterop.godot_callable Godot.NativeInterop.Marshaling.ConvertCallableToNative(Godot.Callable&)
                 NativeCalls.cs:5815 @ void Godot.NativeCalls.godot_icall_3_649(nint, nint, nint, nint, Godot.Callable&)
                 NavigationServer3D.cs:1277 @ void Godot.NavigationServer3D.BakeFromSourceGeometryData(Godot.NavigationMesh, Godot.NavigationMeshSourceGeometryData3D, Godot.Callable)
                 NavigationTest.cs:49 @ void NavigationTest._Ready()
                 Node.cs:2121 @ bool Godot.Node.InvokeGodotClassMethod(Godot.NativeInterop.godot_string_name&, Godot.NativeInterop.NativeVariantPtrArgs, Godot.NativeInterop.godot_variant&)
                 Node3D.cs:1035 @ bool Godot.Node3D.InvokeGodotClassMethod(Godot.NativeInterop.godot_string_name&, Godot.NativeInterop.NativeVariantPtrArgs, Godot.NativeInterop.godot_variant&)
                 NavigationTest_ScriptMethods.generated.cs:78 @ bool NavigationTest.InvokeGodotClassMethod(Godot.NativeInterop.godot_string_name&, Godot.NativeInterop.NativeVariantPtrArgs, Godot.NativeInterop.godot_variant&)
                 CSharpInstanceBridge.cs:24 @ Godot.NativeInterop.godot_bool Godot.Bridge.CSharpInstanceBridge.Call(nint, Godot.NativeInterop.godot_string_name*, Godot.NativeInterop.godot_variant**, int, Godot.NativeInterop.godot_variant_call_error*, Godot.NativeInterop.godot_variant*)
```

### Steps to reproduce

Use NavigationServer3D.BakeFromSourceGeometryData without a callable and with a callable to see the difference also do the same in gdscript (only without the callable parameter).

I'll provide MRP tomorrow or on monday.

### Minimal reproduction project

I'll provide this on a day or two, I'm off right now.

**Comment by Calinou** [comment]
cc @smix8

**Comment by smix8** [comment]
Can't really help with or debug C# specifics since it works with GDScript.

**Comment by raulsntos** [comment]
This is all on C#'s side, we weren't considering "null" Callables when marshaling. https://github.com/godotengine/godot/pull/83357 should fix it.

**Comment by viksl** [comment]
Just to track this also happens with `ParseSourceGeometryData`, I assume the fix will work everywhere just fine so I'm only adding this for the record. Thanks for the fix, hope it makes it into 4.2 soon ;).
